feat: add support for sending sticker packs

sergiooak opened the pull request:
# PR Details

**Summary:**  
Adds support for sending sticker packs, updates typings, and provides usage examples.

## Description

- Introduced `sendMediaAsStickerPack` option to the API and typings.
- Implemented logic for creating and sending sticker packs.
- Updated `example.js` with new commands for sticker and sticker pack sending.
- Added required dependencies: `archiver`, `sharp`.
- Improved type definitions and internal documentation.

## Motivation and Context

This change allows users to send multiple stickers as a pack, improving feature parity with WhatsApp’s native functionality and expanding bot/media capabilities.

## How Has This Been Tested

- Manual testing using the updated `example.js` bot commands (`!sticker`, `!pack`).
- Verified sticker and sticker pack sending in group and private chats.

### Environment

- Machine OS: **Windows 11**
- Phone OS: **Android 14**
- Library Version: **2.3000.1024258380**
- WhatsApp Web Version: **2.3000.1024258380**
- Puppeteer Version: **18.2.1**
- Browser Type and Version: **Chrome 138**
- Node Version: **v22.15.0**

## Types of changes

- [X] Dependency change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly (index.d.ts).
- [X] I have updated the usage example accordingly (example.js)

sergiooak committed:
feat(sticker-packs): add support for sending sticker packs

- Added sendMediaAsStickerPack option to API and typings
- Implemented sticker pack creation and sending logic
- Updated example.js with !sticker and !pack commands
- Added required dependencies: archiver, sharp
- Improved type definitions and

sergiooak committed:
feat(sticker-pack): add support for custom sticker pack thumbnails

sergiooak commented:
<p align="left">
  <img src="https://github.com/user-attachments/assets/ed1008c0-29b0-46ca-acb0-9a056be5dce8" width="200"/>
  <img src="https://github.com/user-attachments/assets/332eac8b-7f74-4d0f-b902-83600154aed0" width="200"/>
  <img src="https://github.com/user-attachments/assets/1e21e047-a80b-44ae-86f4-d54de984f2ab" width="200"/>
</p>

sergiooak commented:
@alechkos About the introduced dependencies, it uses the before optionals archiver and unzipper, and the sharp is used to generate the pack thumbnail.
I guess we can keep the deps as optionals, like the approach on url preview generation, and avoid sharp using another way of generating the png, what do you think:

alechkos reviewed (changes requested):
Please don't use sharp.

sergiooak commented:
Yeah, I figured out why already haha, tomorrow I'll rework to get rid of sharp

sergiooak committed:
refactor: replace sharp with browser-side cropAndResizeImage for sticker pack thumbnails

sergiooak committed:
chore(package): rollback package versions to match base repo state

- Downgraded 'archiver' from ^7.0.1 to ^5.3.1
- Rolled back 'unzipper' from ^0.10.14 to ^0.10.11
- Reverted other version changes introduced in the PR

sergiooak commented:
@alechkos I get rid of sharp, and now I am using `window.WWebJS.cropAndResizeImage` to generate the 64x64px PNG for the sticker-pack thumbnail

alechkos committed:
Merge branch 'main' into main

alechkos committed:
Merge branch 'main' into main

alechkos reviewed (commented):
Hi @sergiooak, I tested and it seems that there is no difference between sending a regular pack and one with a custom thumbnail.

sergiooak commented:
@BenyFilho and @hellhound20236-blip I don't work on this feature in a long time, I'll try to update and test this week

josearg770bis commented:
work ? @sergiooak
